### ISSUE TITLE
AxoCover Project status

### ISSUE THREAD

**Issue by webprofusion-chrisc** [axodox/AxoCover#203]
Firstly, thanks to @axodox for making this great tool available for everyone to use!

It's clear that @axodox is currently spending time away from this project and has perhaps run out of time or motivation (I've certainly experience this on other open projects I have been responsible for).

Comments on issues along the lines of 'Waiting for this' etc are not helpful if the maintainer has reached the limit of their current involvement. Therefore I would propose this issue as a way for capable volunteers to step up and demonstrate their willingness to contribute PRs and to contribute reviews and PR testing.

Please indicate your support for this here.

**Comment by webprofusion-chrisc** [comment]
Re #188

**Comment by malcolmarobertson** [comment]
I support this motion

**Comment by axodox** [comment]
I also support it. 
I can provide technical some help or explain how things were done, but I lack the time to do actual development right now.

**Comment by blkbam** [comment]
I'd be up for that. We've been using it at my company and would like to continue doing so

**Comment by MarkusAmshove** [comment]
@axodox would you be interested in adding maintainers, or if by possible in personal repositories, transferring the repository into an AxoCover organization on github which could in turn have additional maintainers which are chosen in some way?
This way we the extension could live on :-)

**Comment by MABrunner** [comment]
This extension has been invaluable for my company.  I would love to help but I don't know how to write VS extensions.  It would be great to see it continue.  Thanks to @axodox for his excellent work.

**Comment by hector-tn** [comment]
I would love to be part of this since my company values this extension and we actually like using it. DotCover still lacks some of the functionality we can get from AxoCover.

**Comment by axodox** [comment]
@MarkusAmshove Yes, I would be very happy if the community could take over the development, because it would make my work on this have effect on the long term too - even if I no longer have time to participate in coding.

Some info about why I did stop the development:
When I started the project I had a lot of free time and done a lot of testing during work, but found the tools we had quite cumbersome to use, so I started creating this plug-in. In 2-3 weekends I build simple tool which only supported MSTEST (v1) which we used at work. We used it for a while by copying it on flash drives and the local network. Later I put it on the VS gallery so we could update it simpler. A couple of weeks later I noticed some downloads which were certainly not done by my team.

The downloads started to grow and AxoCover was born. Then I kept adding more and more features, almost like going home from my daytime work to a second shift of my own. I added support to xUnit and NUnit (which to be honest I did not used before), and made the UI nicer, and kept adding features I needed or asked by community.

The project quickly started to take more and more time, and after around a year I felt that I can no longer satisfy the community alone, but I also failed to find support locally or on-line. And then I got to do an innovative AR project of my own proposal at work. Since governing a startup like project is very time consuming with demos and finding investors I could not continue active development on the hobby project effectively.  

When I stopped the active work I was thinking on if I could bring in more people or perhaps donate the project in same way to the community (e.g. .Net foundation). Adding testing framework specific features were too big for me to do in my free time, and even keeping up with the VS updates took some significant time. 

At some point I even considered making some way to finance the development, but commercializing it would have gone against my original motivation: making a free code coverage plug-in, which anybody can use be it a professional developer like me, an engineer or a student at a university, Adding a donate button would have been possible but the tax laws in Hungary do not seem to encourage this unless you are sure you will get a certain level of income and get a lawyer to set up the scenario.

So now about one year later I stopped I would be happy if you could help keeping the project alive. I am quite busy but I will try to look into the GitHub organization suggestion in the coming weeks. We would also need to:
- Define some roadmap on what we need the most
- Do some estimation and see if the community would have the time required
- Hold some sessions or document how to set up the development environment
- Provide some technical background on how AxoCover works, and how it could be improved / updated
- Find a way for the community to trigger builds on the Appveyor CI servers.
- Support community to update to development branches (AxoCover allows to select git branches and with Appveyor it allows to update to these versions)
- Of course we would also need to have some oversight on things, like code reviews since we must not break current installations, and make sure that the software does not contain security issues, such as back doors etc.

**Comment by MarkusAmshove** [comment]
Thank you for all the work you've put in  :-)

**Comment by axodox** [comment]
So now we should define, the most needed changes, this would probably include:
- support for current versions of NUnit, XUnit - I think for NUnit this is probably simple, but Xunit was changed a lot under the hood so the adapter for it must be changed too, I would say NUnit would not take long (a day max), Xunit could take a few
- .Net Core support (I have successfully did a try with coverlet) and it worked, but would require quite some effort to make it fully featured, however there are no major technical issues, other than time. That is to make it work nicely it would take a week or even two.
- bugfixes
- The current runner system should be reiterated, as it is a little hacky right now: it uses the existing VS plug-ins of Xunit, NUnit and MSTest, however in the end I needed to fork the code of XUnit and NUnit adapters to make it support all the features. For long term these changes or equivalent should be migrated back to the original repos to make life easier. - So we would not need to embed the plugins themselves, and we could avoid updating AxoCover to support new versions. This would need to have some community support to happen. As some changes I did were already rejected when had been made by others on Xunit for example.

I hope this provides some idea, what I would envision for the project to go forward.
As you might have noticed I put life back into the CI system, and created a VS2019 build yesterday, so I try to do my part :).

**Comment by jgasiorowski** [comment]
Hello, 
Is there any chance that you could look on pending PRs? I've created one for #207 and I guess it can close quite a few other issues related with OpenCover. I know it is not like out-of-the-box solution for problem but it gives ability to users to set their configuration so they don't need to register coverage tool themselves. 
I am asking because one of you bullet points in previous comment is 'bugfixes' so maybe it is worth preparing a release?

(actually I am dying to see that fix in AxoCover - I will really help me) 😛

**Comment by jgasiorowski** [comment]
I am really sad that it seems this project is dead :(
I would be really happy to somehow support this project because I am using it on daily basis. I am not really experienced with Extensions development but I could spend some time on it. But it will be very hard to pull more contributors into it if you will not respond to any requests. Noone would want to spend additional time and waste it

**Comment by jgasiorowski** [comment]
Hello @axodox  I was thinking about trying to add net core support. I was wondering if you wanted to replace OpenCover with coverlet or you wanted to use coverlet only for net core libraries?

**Comment by RoyChase** [comment]
I'd be happy to contribute if there was a set of tasks that were agreed.  .Net core support would be #1 for me but I'm not sure where to start

**Comment by Jsparham777** [comment]
Hi. I would also like to support the project. Is there a back log for this project? Visual Studio 2019, xUnit 2.4.1+ and .NET Core 3.1+ support would be great.

**Comment by lyrray** [comment]
I would like to contribute as well.
